### PR TITLE
Implement RegisteredClaims

### PR DESCRIPTION
**Pull request by lawliet89**

### CODE REVIEW

**Comment by gyng** [comment]
See: https://serde.rs/attributes.html#field-attributes

`#[serde(skip_serializing_if = "path")`

File: src/lib.rs [hunk 1]
```diff
@@ -53,6 +53,190 @@ impl<T> Part for T
...
+    /// Issued at Time in seconds since Unix Epoch
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub iat: Option<u64>,
+    /// JWT ID
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub jti: Option<String>,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct ClaimsSet<T: Serialize + Deserialize> {
```
Comment (gyng) [inline src/lib.rs]: Enum this?

```rust
enum MyClaim {
  RegisteredClaimA,
  RegisteredClaimB,
  Private(T),
}
```

File: src/jws.rs [hunk 2]
```diff
@@ -229,31 +292,46 @@ mod tests {
     /// The base64 encoding will be in `STANDARD` form and not URL_SAFE.
```
Comment (shioju) [inline src/jws.rs]: is this comment correct?

File: src/lib.rs [hunk 3]
```diff
@@ -53,22 +54,108 @@ impl<T> Part for T
...
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+                where E: de::Error
+            {
+                Ok(SingleOrMultipleStrings::Single(s.to_string()))
+            }
+
+            fn visit_string<E>(self, s: String) -> Result<Self::Value, E>
+                where E: de::Error
+            {
+                Ok(SingleOrMultipleStrings::Single(s.to_string()))
```
Comment (shioju) [inline src/lib.rs]: s is already String

File: src/jws.rs [hunk 4]
```diff
@@ -1,27 +1,106 @@
...
 // TODO: Implement verification for registered headers and support custom headers
 // https://tools.ietf.org/html/rfc7515#section-4.1
 pub struct Header {
     pub alg: Algorithm,
     /// Type of the JWT. Usually "JWT".
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub typ: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub jku: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (lawliet89) [inline src/jws.rs]: Already tested

**Pull request merged**